frontend/src/pages/HomePage/HomePage.js

```diff
@@ -1,91 +1,209 @@
+                            username: item.creatorAccountInfo.username,
+                            followed: userData ? userData.following.includes(item.creatorAccountInfo.username) : false,
+                            imageId: item.creatorAccountInfo.profilePictureId,
+                        },
+                        content: {
+                            type: "event",
+                            id: item.id,
+                            title: item.eventInfo.title,
+                            description: item.eventInfo.description,
+                            imageId: item.eventInfo.posterId,
+                            creationDate: item.creationDate,
+                        }
+                    }
+
+                    setContent(content => [...content, event])
+                });
+
+                setLoaded(true);
+                // setContent((c) => c.sort((a, b) => (a.content.creationDate > b.content.creationDate) ? -1 : 1));
+
+            })
+            .catch((error) => {
                 setLoaded(true)
-                setEvents(data)
-            },
-                error => {
-                    setLoaded(true)
-                    setError(error)
-                })
-    }, [token])
+                setError(error)
+            })
+
+        fetch('api/discussionPost', fetchArgs)
+            .then((response) => response.json())
+            .then((data) => {
+                data.forEach((item) => {
+
+                    const discussionPost = {
+                        creator: {
+                            id: item.creatorAccountInfo.id,
+                            username: item.creatorAccountInfo.username,
+                            followed: userData ? userData.following.includes(item.creatorAccountInfo.username) : false,
+                            imageId: item.creatorAccountInfo.profilePictureId,
+                        },
+                        content: {
+                            type: "discussionPost",
+                            id: item.id,
+                            title: item.title,
+                            description: item.textBody,
+                            creationDate: item.creationDate,
+                        }
+                    }
+
+                    setContent(content => [...content, discussionPost])
+                });
+
+                setLoaded(true);
+
+            })
+            .catch((error) => {
+                setLoaded(true);
+                setError(error)
+            })
+
+
+    }, [token, userData])
+
+
+    // TODO: Sort by date
+    // useEffect(() => {
+    //     const sortedContent = content.sort((a, b) => (a.content.creationDate > b.content.creationDate) ? -1 : 1);
+    //     setContent(sortedContent);
+    // }, [isLoaded])
 
     if (error) {
         return <div>Error: {error.message}</div>
     } else if (!isLoaded) {
-        return <div>Loading...</div>
+        return <div><CircularProgress /></div>
```

Comment:
Should I add this to the other pages if it looks better?

Reply:
I don't think that it is necessary for now